cajita/src/Cajita_SparseIndexSpace.hpp

```diff
@@ -0,0 +1,564 @@
+#ifndef CAJITA_SPARSE_INDEXSPACE_HPP
+#define CAJITA_SPARSE_INDEXSPACE_HPP
+
+#include <Kokkos_Atomic.hpp>
+#include <Kokkos_Core.hpp>
+#include <Kokkos_UnorderedMap.hpp>
+
+#ifdef __CUDACC__
+#define CUDA_HOSTDEV __host__ __device__
+#else
+#define CUDA_HOSTDEV
+#endif
+
+#include <array>
+#include <string>
+
+namespace Cajita
+{
+
+enum class HashTypes : unsigned char
+{
+    Naive = 0,
+    Morton = 1
+};
+
+//---------------------------------------------------------------------------//
+// bit operations for Morton Code computing
+template <typename Integer>
+CUDA_HOSTDEV constexpr Integer bit_length( Integer N ) noexcept
```

Comment:
For functions we use camel case starting with lower case. So this function would be called `bitLength()`. Please change the naming conventions for all functions.